### ISSUE TITLE
ExecuteSync throws timeout-error when called from inside a task

### ISSUE THREAD

**Issue by steininger** [StackExchange/StackExchange.Redis#460]
I think ExecuteSync can not be run from codepath started by a task. it returns imediatly a timeout.
Used by all sync commands like SetMembers, .. (vs SetMembersAsync).
I investigated the following: it uses a kind of task in the backgroud for witch is waited. so when the calling code path is also a path the outside task returns the await when the inside task only start. normaly this is avoided by `.Unwrap()` ... Maybe change (https://github.com/StackExchange/NetGain/blob/4098b88493f560745ffb9b0980ae8b0068d463fe/StackExchange.NetGain/TcpClientGroup.cs#L183) to 

``` C#
        public object ExecuteSync(object message, int timeout = -1)
        {
            var task = Task.Factory.StartNew(async () => await Execute(message, true, null)).Unwrap();
            if(task.Wait(timeout))
            {
                return task.Result;
            }
            else
            {
                throw new Exception("timeout");
            }
        }
```

as an example ....
i havn't took a look into you own Wait(task,timeout) method, maybe it is already implemented there.

why call a sync method inside async code path: using cache inside a constructor of object created in async code path ....

if needed i can surfe a detailed example.

**Comment by rossipedia** [comment]
what version of .net are you targeting?

**Comment by NickCraver** [comment]
Closing this out to cleanup, it should be resolved by the continuation changes in 1.1.605 and later.
